pkg/parquetquery/iters_test.go

```diff
@@ -43,21 +43,68 @@ func TestNext(t *testing.T) {
 	}
 }
 
+// TestTruncate compares the unrolled TruncateRowNumber() with the original truncateRowNumberSlow() to
+// prevent drift
+func TestTruncateRowNumber(t *testing.T) {
+	for i := 0; i < 1000; i++ {
+		rn := RowNumber{1, 2, 3, 4, 5, 6, 7, 8}
+		d := rand.Intn(MaxDefinitionLevel + 1)
+
+		newR := TruncateRowNumber(d, rn)
+		oldR := truncateRowNumberSlow(d, rn)
+
+		require.Equal(t, newR, oldR)
+	}
+}
+
+func TestInvalidDefinitionLevel(t *testing.T) {
```

Comment:
`nitpick` It is a bit unusual to test two different and only slightly related functions in the same test. It would be more idiomatic to split this in two tests `TestTruncateRowNumber_InvalidLevel` and `TestRowNumber_InvalidLevel (or similar names). Or as an alternative, add them as sub test to `TestTruncateRowNumber` and `TestRowNumber`